kirovy/models/cnc_game.py

```diff
@@ -3,10 +3,67 @@
 from django.conf import settings
 from django.db import models
 
-from kirovy import exceptions
+from kirovy import exceptions, typing as t
 
 from kirovy.models.cnc_base_model import CncNetBaseModel
 
+__all__ = ["CncFileExtension", "CncGame"]
+
+
+def is_valid_extension(extension: str) -> None:
+    """Validate file extension string.
+
+    :raise exceptions.InvalidFileExtension:
+        Raised for invalid file extension strings.
+    """
+    if not extension.isalnum():
+        raise exceptions.InvalidFileExtension(
+            f'"{extension}" is not a valid file extension. Must be alpha only.'
+        )
+
+
+class CncFileExtension(CncNetBaseModel):
+    """File extension types for Command & Conquer games and what they do.
+
+    Useful page: https://modenc.renegadeprojects.com/File_Types
+    """
+
+    class ExtensionTypes(models.TextChoices):
+        """Enum of what kind of file this is to allow for filtering allowed extensions."""
+
+        MAP = "map", "map"
+        """This file extensions represents some kind of map file and should be in the map database."""
+
+        ASSETS = "assets", "assets"
+        """This file extension represents some kind of game asset to support a map, e.g. a ``.mix`` file."""
```

Comment:
I plan to only do the map db right now, I'm just getting the infrastructure ready in advance so that it doesn't require a huge data migration later on

Reply:
So you mean that it would be in the same service? That would probably make sense yeah.